src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -543,11 +582,20 @@ namespace OpenMS
       const Int x = act_coords[0], y = act_coords[1];
 
       OpenMS::GridFeature* center_feature = it->second;
-      QTCluster cluster(center_feature, num_maps_, max_distance, use_IDs_, x, y);
+
+      // construct empty data body for the new cluster and create the head afterwards
+      cluster_data.emplace_back();
+      
+      QTCluster cluster(&cluster_data.back(), center_feature, num_maps_, 
+                                        max_distance, use_IDs_, x, y, id);
 
       addClusterElements_(x, y, grid, cluster, center_feature);
 
-      clustering.push_back(cluster);
+      // push the cluster head of the new cluster into the heap
+      handles[id] = cluster_heads.push(cluster);
+
+      // next cluster gets the next id
+      ++id;
```

Comment:
Can move this into the loop-definition. If you need the comment, leave it here.

Reply:
I prefer it this way.